src/gluonts/nursery/temporal_hierarchical_forecasting/eval/evaluation.py

```diff
@@ -87,27 +88,41 @@ def evaluate_forecasts_at_all_levels(
     evaluator,
     metrics: List[str] = ["mean_wQuantileLoss"],
 ):
+    # (ts_1_6M, ts_1_2M, ts_1_1M, ts_2_6M, ts_2_2M, ts_2_1M, ...)
     forecast_at_all_levels_unpacked_it = unpack_forecasts(
         forecast_at_all_levels_it=forecast_at_all_levels_it,
         temporal_hierarchy=temporal_hierarchy,
         target_temporal_hierarchy=temporal_hierarchy,
     )
 
-    # First get item metrics for all time series for all frequencies; these are per time series metrics.
```

Comment:
I would suggest to keep a message like this one, as it explains why we are using `tee` in this particular way.

Reply:
Ya, added comments. Thanks!